core/src/main/scala/org/http4s/Method.scala

```diff
@@ -54,54 +53,97 @@ object Method extends MethodInstances {
-  val `VERSION-CONTROL` = register(new Method("VERSION-CONTROL") with Idempotent)
+  val ACL: IdempotentMethod = new Method("ACL") with Idempotent
+  val `BASELINE-CONTROL`: IdempotentMethod = new Method("BASELINE-CONTROL") with Idempotent
+  val BIND: IdempotentMethod = new Method("BIND") with Idempotent
+  val CHECKIN: IdempotentMethod = new Method("CHECKIN") with Idempotent
+  val CHECKOUT: IdempotentMethod = new Method("CHECKOUT") with Idempotent
+  val CONNECT: DefaultMethodNoBody = new Method("CONNECT") with Default with NoBody
+  val COPY: IdempotentMethod = new Method("COPY") with Idempotent
+  val DELETE: IdempotentMethodNoBody = new Method("DELETE") with Idempotent with NoBody
+  val GET: SafeMethodNoBody = new Method("GET") with Safe with NoBody
+  val HEAD: SafeMethodNoBody = new Method("HEAD") with Safe with NoBody
+  val LABEL: IdempotentMethodWithBody = new Method("LABEL") with Idempotent with PermitsBody
+  val LINK: IdempotentMethod = new Method("LINK") with Idempotent
+  val LOCK: DefaultMethod = new Method("LOCK") with Default
+  val MERGE: IdempotentMethod = new Method("MERGE") with Idempotent
+  val MKACTIVITY: IdempotentMethod = new Method("MKACTIVITY") with Idempotent
+  val MKCALENDAR: IdempotentMethod = new Method("MKCALENDAR") with Idempotent
+  val MKCOL: IdempotentMethod = new Method("MKCOL") with Idempotent
+  val MKREDIRECTREF: IdempotentMethod = new Method("MKREDIRECTREF") with Idempotent
+  val MKWORKSPACE: IdempotentMethod = new Method("MKWORKSPACE") with Idempotent
+  val MOVE: IdempotentMethod = new Method("MOVE") with Idempotent
+  val OPTIONS: SafeMethodWithBody = new Method("OPTIONS") with Safe with PermitsBody
+  val ORDERPATCH: IdempotentMethod = new Method("ORDERPATCH") with Idempotent
+  val PATCH: DefaultMethodWithBody = new Method("PATCH") with Default with PermitsBody
+  val POST: DefaultMethodWithBody = new Method("POST") with Default with PermitsBody
+  val PROPFIND: SafeMethod = new Method("PROPFIND") with Safe
+  val PROPPATCH: IdempotentMethod = new Method("PROPPATCH") with Idempotent
+  val PUT: IdempotentMethodWithBody = new Method("PUT") with Idempotent with PermitsBody
+  val REBIND: IdempotentMethod = new Method("REBIND") with Idempotent
+  val REPORT: SafeMethod = new Method("REPORT") with Safe
+  val SEARCH: SafeMethod = new Method("SEARCH") with Safe
+  val TRACE: SafeMethodWithBody = new Method("TRACE") with Safe with PermitsBody
+  val UNBIND: IdempotentMethod = new Method("UNBIND") with Idempotent
+  val UNCHECKOUT: IdempotentMethod = new Method("UNCHECKOUT") with Idempotent
+  val UNLINK: IdempotentMethod = new Method("UNLINK") with Idempotent
+  val UNLOCK: IdempotentMethod = new Method("UNLOCK") with Idempotent
+  val UPDATE: IdempotentMethod = new Method("UPDATE") with Idempotent
+  val UPDATEREDIRECTREF: IdempotentMethod = new Method("UPDATEREDIRECTREF") with Idempotent
+  val `VERSION-CONTROL`: IdempotentMethod = new Method("VERSION-CONTROL") with Idempotent
+
+  val all = List(
```

Comment:
This is going to be easy to forget to update, but that happens basically never.

Reply:
Right, that's a limitation. We could do an automated tests that loads the spec on xml and verifies but it seems too much effort

Reply:
Another automated test could be done through reflection, but that test wouldn't run on scala.js, which is part of the point.